csharp/src/Apache.Arrow/Arrays/BinaryArray.cs

```diff
@@ -164,21 +164,21 @@ public TBuilder Clear()
 
         public ArrowBuffer ValueBuffer => Data.Buffers[2];
 
-        public ReadOnlySpan<int> ValueOffsets => ValueOffsetsBuffer.Span.CastTo<int>().Slice(0, Length + 1);
+        public ReadOnlySpan<int> ValueOffsets => ValueOffsetsBuffer.Span.CastTo<int>().Slice(Offset, Length + 1);
 
         public ReadOnlySpan<byte> Values => ValueBuffer.Span.CastTo<byte>();
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public int GetValueOffset(int index)
         {
-            return ValueOffsets[Offset + index];
+            return ValueOffsets[index];
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public int GetValueLength(int index)
         {
             var offsets = ValueOffsets;
-            var offset = Offset + index;
+            var offset = index;
```

Comment:
Can we remove this variable and use `index` directly?

Reply:
I modified.